### PR TITLE
Performance test for vhost-user-blk

### PR DESCRIPTION
**Pull request by kalyazin**
## Changes

This adds a performance test for the vhost-user-blk device.

## Reason

vhost-user-blk device is added in https://github.com/firecracker-microvm/firecracker/pull/4170 .
We need to include performance tests for the new device.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

~- [ ] If a specific issue led to this PR, this PR closes the issue.~
- [x] The description of changes is clear and encompassing.
~- [ ] Any required documentation changes (code and docs) are included in this PR.~
~- [ ] API changes follow the [Runbook for Firecracker API changes][2].~
~- [ ] User-facing changes are mentioned in `CHANGELOG.md`.~
- [x] All added/changed functionality is tested.
~- [ ] New `TODO`s link to an issue.~
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/integration_tests/performance/test_block_vhost_user_ab.py
Comment (roypat) [inline tests/integration_tests/performance/test_block_vhost_user_ab.py:1]: I think we can just add this to `test_block_ab.py`, since it imports all the utility functions from there anyway. Similar to how the network latency and network throughput tests both live in `test_network_ab.py` (maybe some of the vm setup can also be extracted into a helper function, similar to what's done in test_network_ab.py, too)

Comment (kalyazin) [reply tests/integration_tests/performance/test_block_vhost_user_ab.py:1]: yes, good point. Will see what it looks like.

Comment (kalyazin) [reply tests/integration_tests/performance/test_block_vhost_user_ab.py:1]: done

**Review by roypat — APPROVED** [review]

File: tests/host_tools/drive.py [hunk 1]
```diff
@@ -19,6 +20,14 @@ def __init__(self, path: str, size: int = 256, fs_format: str = "ext4"):
         Raises if the file system format is not supported, if the file already
         exists, or if it ends in '/'.
         """
+
+        # If no path is supplied, use a temporary file.
+        # This is useful to force placing the file on disk, not in memory,
+        # because qemu vhost-user-blk backend always uses O_DIRECT,
+        # but O_DIRECT is not supported by tmpfs.
+        if path is None:
```
Comment (bchalios) [inline tests/host_tools/drive.py:28]: It is worth making the `path` parameter in `__init__` optional, like: `path: str = None`

File: .buildkite/pipeline_ab.py [hunk 2]
```diff
@@ -15,11 +15,17 @@
...
+        "label": "🖴 Virtio Block Performance",
         "test_path": "integration_tests/performance/test_block_ab.py",
         "devtool_opts": "-c 1-10 -m 0",
     },
+    "vhost-user-block": {
+        "label": "🖴 vhost-user Block Performance",
+        "test_path": "integration_tests/performance/test_block_vhost_user_ab.py",
+        "devtool_opts": "-c 1-10 -m 0",
+        "timeout_in_minutes": 40,
+    },
```
Comment (roypat) [inline .buildkite/pipeline_ab.py:28]: This needs to be 

```suggestion
    "virtio-block": {
        "label": "🖴 Virtio Block Performance",
        "test_path": "integration_tests/performance/test_block_ab.py::test_block_performance",
        "devtool_opts": "-c 1-10 -m 0",
    },
    "vhost-user-block": {
        "label": "🖴 vhost-user Block Performance",
        "test_path": "integration_tests/performance/test_block_ab.py::test_block_vhost_user_performance",
        "devtool_opts": "-c 1-10 -m 0",
        "timeout_in_minutes": 40,
    },
```

now

**Review by roypat — CHANGES_REQUESTED** [review]

File: .buildkite/pipeline_ab.py [hunk 3]
```diff
@@ -15,11 +15,17 @@
...
+    "virtio-block": {
+        "label": "🖴 Virtio Block Performance",
         "test_path": "integration_tests/performance/test_block_ab.py",
         "devtool_opts": "-c 1-10 -m 0",
     },
+    "vhost-user-block": {
+        "label": "🖴 vhost-user Block Performance",
+        "test_path": "integration_tests/performance/test_block_vhost_user_ab.py",
+        "devtool_opts": "-c 1-10 -m 0",
+        "timeout_in_minutes": 40,
```
Comment (bchalios) [inline .buildkite/pipeline_ab.py:27]: Why the timeout?

**Review by bchalios — COMMENTED** [review]
TLDR, apart from a nit

File: .buildkite/pipeline_ab.py (hunk 3 above)
Comment (roypat) [reply .buildkite/pipeline_ab.py:27]: I think these are rebase artifacts, good catch. We removed the timeouts from the pipeline steps a while ago, since we rely on the per-test pytest timeouts here.

File: .buildkite/pipeline_ab.py (hunk 2 above)
Comment (kalyazin) [reply .buildkite/pipeline_ab.py]: indeed, thanks!

File: .buildkite/pipeline_ab.py (hunk 3 above)
Comment (kalyazin) [reply .buildkite/pipeline_ab.py]: fixed as a part of Patrick's comment.

File: tests/host_tools/drive.py (hunk 1 above)
Comment (kalyazin) [reply tests/host_tools/drive.py:28]: true, done, thanks

**Review by bchalios — APPROVED** [review]

**Review by roypat — APPROVED** [review]

File: .buildkite/pipeline_ab.py (hunk 2 above)
Comment (roypat) [reply .buildkite/pipeline_ab.py]: Sorry for not catching this earlier >.>
